### PR TITLE
Fix the vulnerability for public method that return private type

### PR DESCRIPTION
**Pull request by yugeeklab**
Change public method that return private type to public method that
return copy.

Signed-off-by: yugeeklab <yugeeklab@gmail.com>

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Documentation update
- [x] Refactoring, Maintenance
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### CODE REVIEW

File: src/main/java/oss/fosslight/domain/BinaryMaster.java [hunk 1]
```diff
@@ -196,13 +196,23 @@ public void setBatWatcher(List<BatWatcher> batWatcher) {
 		this.batWatcher = batWatcher;
 	}
 	public String[] getWatcherDivision() {
-		return watcherDivision;
+		String[] ret = null;
+		if(this.watcherDivision != null) {
+			ret = this.watcherDivision.clone();
+		}
+
+		return ret;
```
Comment (riyenas0925) [inline src/main/java/oss/fosslight/domain/BinaryMaster.java:204]: wouldn't it be better to change it like below?

```java
return watcherDivision.clone();
```

Comment (yugeeklab) [reply src/main/java/oss/fosslight/domain/BinaryMaster.java]: Done!

watcherDivision != null ? watcherDivision.clone() : null;

Thank you for your review!

Comment (riyenas0925) [reply src/main/java/oss/fosslight/domain/BinaryMaster.java]: Is there any reason to do a null check?

Comment (yugeeklab) [reply src/main/java/oss/fosslight/domain/BinaryMaster.java]: I think (null).clone() will cause null point exception.

Is there no case that watcherDivision can be a null?

Comment (riyenas0925) [reply src/main/java/oss/fosslight/domain/BinaryMaster.java]: I agree :smile:

**Review by soimkim — APPROVED** [review]

File: src/main/java/oss/fosslight/domain/BinaryMaster.java (hunk 1 above)
Comment (soimkim) [reply src/main/java/oss/fosslight/domain/BinaryMaster.java]: It is null checked, so I resolve this conversation.

**Pull request merged**